### PR TITLE
Detect moves in progress and return HTTP 409/ERR_CONFLICT

### PR DESCRIPTION
**Pull request by ianmcorvidae**

### CODE REVIEW

File: src/data_info/services/rename.clj [hunk 1]
```diff
@@ -19,6 +21,42 @@
...
+  ;; statuses registered, started, running, detected-stalled (so, not completed/failed mostly)
+  ;; or we can just only include non-completed ones, i.e. EndDateSince sometime in the future + include null end
+  ;; Pull out their paths.
+  ;; data-move: sources, destination
+  ;; data-rename: source, destination
+  ;; data-delete: paths, trash-paths (not that the latter is real likely)
+  ;; data-delete-trash: trash-paths
+  ;; data-restore: paths, restoration-paths
+  ;; Then we need to check all of `paths` to see if any of them matches any of the paths we fetched. These should either match exactly, or match something up through a slash in the path passed to us (i.e., a folder -- we don't want to lock /x/y/zarbee if someone is moving /x/y/za to /x/y/zb or whatever)
+  (let [far-future "9999-12-31T23:59:59Z"
```
Comment (johnworth) [inline src/data_info/services/rename.clj:37]: Glad I won't be around for the inevitable Y10K apocalypse. 😁

**Comment by ianmcorvidae** [comment]
https://github.com/cyverse-de/clojure-commons/commit/fbed6641240a3e8849c38539e5b5aecc3307570a and https://github.com/cyverse-de/async-tasks-client/commit/ee1a8111af090f79cfeac36cca17ed1e729aaf77 are both part of this change as well. Neither should affect anything else.

**Review by johnworth — APPROVED** [review]
I'm not seeing any obvious issues. Merging and getting it into QA seems good to me.

File: src/data_info/services/rename.clj (hunk 1 above)
Comment (ianmcorvidae) [reply src/data_info/services/rename.clj:37]: I figure, if anyone is still running this code unmodified at that point, they have bigger problems :laughing:

**Review by psarando — APPROVED** [review]
LGTM 👍 

I don't see async-tasks getting called from anywhere else, so hopefully this covers all the cases.

File: src/data_info/services/rename.clj [hunk 2]
```diff
@@ -19,6 +21,42 @@
...
+        eligible-async-task-types ["data-move" "data-rename" "data-delete" "data-delete-trash" "data-restore"]
+        eligible-tasks (async-tasks/get-by-filter {:type eligible-async-task-types
+                                                   :include_null_end true
+                                                   :end_date_since far-future})
+        extract-paths (fn [task]
```
[the remark this answers is not in the document]
Comment (ianmcorvidae) [reply src/data_info/services/rename.clj:42]: Oh yeah, that's a good idea. I always forget to do that sort of thing, thanks for the reminder!

**Comment by ianmcorvidae** [comment]
Alright, I think this is good enough to put in QA. I'll try to break it in QA later tonight or tomorrow.

**Pull request merged**